website/addons/base/views.py

```diff
@@ -223,6 +227,42 @@ def get_or_start_render(file_guid, start_render=True):
     return None
 
 
+def file_content(file_guid):
+    content = None
+    if is_editable(file_guid):
+        content = get_file_contents(
+            file_guid.mfr_download_url,
+            file_guid.mfr_cache_path,
+            file_guid.mfr_temp_path,
+            file_guid.public_download_url,
+        )
+
+    return content
+
+
+def is_editable(file_guid):
+    try:
+        file_guid.enrich()
+    except exceptions.AddonEnrichmentError:
+        return False
+
+    file_type = get_mimetype(file_guid.waterbutler_path)
+    if file_type is not None:
```

Comment:
This works for now, but I'd rather see a constant defined somewhere that contains the list of editable files, i.e. something like:
```EDITABLE_FILETYPES = [ftype for ftype in FILE_TYPES if ftype.split('/')[0] == 'text']```
and then:
``` if file_type in EDITABLE_FILETYPES:```
If we extend the file editor in the future to support more file types it will be better to just change the constant.